### ISSUE TITLE
Model in partial template

### ISSUE THREAD

**Issue by azjol** [salesforce/handlebars-php#4]
Hello, 
I cannot send data into partial template

`{{> partials/_head title="Header title" }}`

Is it possible or Am I doing something wrong?
Thank you

**Comment by michaelphipps** [comment]
Hi @azjol

# tldr;
Your partial reference in the main template should look like this
```handlebars
{{> _head data }}
```
Your partial would contain this:
```handlebars
<h1>{{title}}</h1>
```
and you pass the data via `render()` this way
```php
$handlebars->render("main", [
    "data"=>[
        "title"=>"Header title"
        ]
    ]
);
```
Read on for more detail.  Sorry it is long.  This info is missing or isn't clear in the current docs.

## Partial Naming
You can't refer to directories in the partial's name.  You can only use the name of the file.  

If you do want to put partials in a different directory to templates, you need to set your code up like this:
```php
# assuming you've already set up $templatesLoader ;)
...
# Set the partial files
$partialsDir = __DIR__."/templates/partials";
$partialsLoader = new FilesystemLoader($partialsDir, [
        "extension" => "partial"
    ]
);

# Instantiate Handlebars
$handlebars = new Handlebars([
    "loader" => $templatesLoader,
    "partials_loader" => $partialsLoader,
    
]);
...
```
Note: I prefer to keep my partials in the same folder as templates, so I give them the extension `.partial`

## Passing Data
The way you pass data to partials works differently in this php library than it does in handlebars.js.

Assuming the following setup:

main.tpl
```handlebars
{{> _head}}
```
_head.partial
```handlebars
<h1>{{title}}</h1>
```
If you load the template using
```php
$handlebars->render("main", ["title"=>"Header title"]);
```
the output will be:
```html
<h1>Header title</h1>
```
Because the partial is aware of the data available to the parent.

The moment you pass data to the partial, it only knows about the data passed to it.  If you changed main.tpl to:
```handlebars
{{> _head title}}
```
the output will be:
```html
<h1></h1>
```
`title` actually contains the string `"Header title"` and not an array containing `title` so there is no `{{title}}` value it can insert.

There is a special `{{.}}` reference that contains the context (the data that was sent to the partial) that can come in handy!  If you change _head.partial to:
```handlebars
<h1>{{.}}</h1>
```
the output will be:
```html
<h1>Header title</h1>
```
To prevent this answer getting too much longer, I've created this gist [How to pass data to partials in Handlebars.php](https://gist.github.com/michaelphipps/c5db7418835e837950dcc34bb757256d) with a better use case for the {{.}}.

**Comment by gitpo** [comment]
Thanks @michaelphipps this is excellent documentation that and would be a great addition to the README. Would you mind spinning up a PR to add this in?

**Comment by theodorejb** [comment]
Consider trying [PHP Handlebars](https://github.com/devtheorem/php-handlebars), a modern project focused on high performance and full Handlebars spec support. It allows passing data to partials exactly [the same way as in Handlebars.js](https://handlebarsjs.com/guide/partials.html#partial-parameters):

```php
use DevTheorem\Handlebars\Handlebars;

require 'vendor/autoload.php';

$template = Handlebars::compile('{{> partials/_head title="Header title" }}');

$partials = [
    'partials/_head' => '<h1>{{title}}</h1>',
];

echo $template([], [
    'partials' => array_map(fn($p) => Handlebars::compile($p), $partials),
]);
```

Output: `<h1>Header title</h1>`
